Rel8/Internal/Table.hs

```diff
@@ -181,6 +184,60 @@ f $? MaybeTable _ x = toNullable (f x)
 isTableNull :: MaybeTable a -> Expr Bool
 isTableNull (MaybeTable tag _) = nullable (lit True) (\_ -> lit False) tag
 
+
+--------------------------------------------------------------------------------
+-- | A pair of 'Table's where at most one might be @null@. This is the result of
+-- an @FULL OUTER JOIN@ between tables.
+data TheseTable a b = TheseTable (MaybeTable a) (MaybeTable b)
+  deriving (Functor)
```

Comment:
It's a shame that the data type doesn't actually enforce the documented invariant, but it's not exported so it's probably not worth worrying about anything more exact.

Reply:
Yeah, it is, but I think we're stuck with something like this anyway because Opaleye's `fullJoin` returns a tuple.